VideoAPI/VideoApi.DAL/Commands/UpdateParticipantStatusAndRoomCommand.cs

```diff
@@ -37,23 +40,46 @@ public UpdateParticipantStatusAndRoomCommandHandler(VideoApiDbContext context)
         public async Task Handle(UpdateParticipantStatusAndRoomCommand command)
         {
             var conference = await _context.Conferences.Include(x => x.Participants)
+                .ThenInclude(x => x.CurrentVirtualRoom).ThenInclude(x => x.RoomParticipants)
                 .SingleOrDefaultAsync(x => x.Id == command.ConferenceId);
-            
+
             if (conference == null)
             {
                 throw new ConferenceNotFoundException(command.ConferenceId);
             }
+            
+            var virtualRoom = await _context.Rooms.SingleOrDefaultAsync(x => x.Label == command.RoomLabel);
 
+            if (!command.Room.HasValue && virtualRoom == null && command.ParticipantState != ParticipantState.Disconnected)
+            {
+                throw new RoomNotFoundException(command.RoomLabel);
+            }
+            
             var participant = conference.GetParticipants().SingleOrDefault(x => x.Id == command.ParticipantId);
             if (participant == null)
             {
                 throw new ParticipantNotFoundException(command.ParticipantId);
             }
-            
+
+            UpdateRoom(command.ParticipantState, virtualRoom, participant);
             participant.UpdateParticipantStatus(command.ParticipantState);
             participant.UpdateCurrentRoom(command.Room);
-            
+            participant.UpdateCurrentVirtualRoom(virtualRoom);
             await _context.SaveChangesAsync();
         }
+
+        private void UpdateRoom(ParticipantState status, Room vRoom, Participant participant)
+        {
+            var isDynamicConsultationRoom = vRoom != null;
+            if (status == ParticipantState.InConsultation && isDynamicConsultationRoom)
+            {
+                vRoom.AddParticipant(new RoomParticipant(vRoom.Id, participant.Id));
```

Comment:
why do i need vroomid? when adding to vroom

Reply:
same below

Reply:
Not anymore. I've updated the domain so this is no longer a requirement